### ISSUE TITLE
Crash when h-local pseudopotential is used.

### ISSUE THREAD

**Issue by mcbennet** [QMCPACK/qmcpack#2917]
In the attached run, QMCPACK crashes when the Bi (h-local) potential is used.

The error reads as follows:
```
[or-condo-c272:113388:0:113436] Caught signal 11 (Segmentation fault: Sent by the kernel at address (nil))
*** Error in `/nfs/home/8gb/Codes/src/QMCPACK/build_cades_cpu_comp_skylake_v3.10.0/bin/qmcpack': corrupted double-linked list: 0x0000000005acbb40 ***
```

When a g-local potential is used instead, QMCPACK runs successfully.

**To Reproduce**
1. release version: v3.10.0
2. `cmake -DENABLE_SOA=1 -DQMC_COMPLEX=1 -DCMAKE_PREFIX_PATH="$HDF5_DIR" $CMAKE_FLAGS ..`
3. `mpirun -np 4 /nfs/home/8gb/Codes/src/QMCPACK/build_cades_cpu_comp_skylake_v3.10.0/bin/qmcpack vmc.in.xml`

**System:**
 - ORNL CADES
 - modules loaded
```
Currently Loaded Modulefiles:
  1) python/3.6.3           2) intel/19.0.3           3) openmpi/3.1.5          4) PE-intel/3.0           5) gcc/7.2.0              6) hdf5-parallel/1.8.21   7) fftw/3.3.5             8) cmake/3.12.0           9) boost/1.70.0          10) libxml2/2.9.9
```

**Additional context**
Both failed and successful runs are attached for h-local and g-local, respectively. Disregard coarse meshfactor -- this system is a bit of a memory hog and will need further work to reduce its memory requirements.

[issue.tar.gz](https://github.com/QMCPACK/qmcpack/files/5982525/issue.tar.gz)

**Comment by prckent** [comment]
Can you paste the actual error here? Where does the program crash?

(FYI, ENABLE_SOA is not needed any more)

**Comment by mcbennet** [comment]
> Can you paste the actual error here? Where does the program crash?

Error printout added to original post.

**Comment by camelto2** [comment]
Closed by #2920
